acai-ios/MenuTabView.swift

```diff
@@ -0,0 +1,153 @@
+        super.init(frame: .zero)
+
+        backgroundColor = .clear
+
+        stackView = UIStackView()
+        stackView.axis = .horizontal
+        stackView.alignment = .fill
+        stackView.distribution = .equalSpacing
+
+        buttons = []
+        self.items = items
+
+        items.enumerated().forEach { arg in
+            let (i, _) = arg
+            let button = UIButton()
+            button.tag = i
+            button.addTarget(self, action: #selector(didSelect), for: .touchUpInside)
+            stackView.addArrangedSubview(button)
+            buttons.append(button)
+        }
+
+        setButtonTitleAttributes()
+
+        underline = UIView()
+        underline.backgroundColor = .mango
+
+        addSubview(stackView)
+        addSubview(underline)
+
+        setUpConstraints()
+    }
+
+    private func setUpConstraints() {
+
+        stackView.snp.makeConstraints { make in
+            make.top.bottom.equalToSuperview()
+            make.leading.trailing.equalToSuperview().inset(contentInset)
+        }
+
+        underline.snp.makeConstraints { make in
+            make.height.equalTo(underlineHeight)
+            make.leading.trailing.equalTo(stackView.subviews[selectedSegmentIndex])
+            make.centerY.equalTo(stackView).offset(getTrueUnderlineOffset())
+        }
+
+    }
+
+    private func remakeContentInset() {
+        guard underline != nil && stackView != nil else { return }
+
+        stackView.snp.remakeConstraints { make in
+            make.top.bottom.equalToSuperview()
+            make.leading.trailing.equalToSuperview().inset(contentInset)
+        }
+
+        layoutSubviews()
+
+        remakeUnderlineConstraints()
+    }
+
+    private func getTrueUnderlineOffset() -> CGFloat {
+        let font: UIFont = titleAttributes[.font] as? UIFont ?? UIFont.systemFont(ofSize: 12)
+        let textHeight = "Test".height(withConstrainedWidth: UIScreen.main.bounds.width, font: font)
+        return textHeight / 2 + underlineOffset
+    }
+
+    private func remakeUnderlineConstraints() {
+        guard underline != nil && stackView != nil else { return }
+
+        underline.snp.remakeConstraints { make in
+            make.height.equalTo(underlineHeight)
+            make.leading.trailing.equalTo(stackView.subviews[selectedSegmentIndex])
+            make.centerY.equalTo(stackView).offset(getTrueUnderlineOffset())
+        }
+    }
+
+    private func setButtonTitleAttributes() {
+        buttons.enumerated().forEach { arg in
+            let (i, button) = arg
+            let attrStr = NSAttributedString(string: items[i], attributes: titleAttributes)
+            button.setAttributedTitle(attrStr, for: .normal)
+        }
+    }
+
+    @objc func didSelect(sender: UIButton) {
+        selectedSegmentIndex = sender.tag
+        sendActions(for: .valueChanged)
+        UIView.animate(withDuration: 0.25) {
+            self.remakeUnderlineConstraints()
+            self.layoutSubviews()
```

Comment:
From apple docs (https://developer.apple.com/documentation/uikit/uiview/1622482-layoutsubviews) I don't think you should call `layoutSubviews()` directly. You can call `layoutIfNeeded()` if you want it to refresh immediately

Reply:
Okay that's what I thought. I took this from something we did in Ceres but didn't know why this was called.